test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -322,24 +326,23 @@ u256 EVMInstructionInterpreter::eval(
 		return (0xdddddd + arg[1]) & u256("0xffffffffffffffffffffffffffffffffffffffff");
 	case Instruction::CALL:
 	case Instruction::CALLCODE:
-		// TODO assign returndata
 		accessMemory(arg[3], arg[4]);
 		accessMemory(arg[5], arg[6]);
 		logTrace(_instruction, arg);
-		return arg[0] & 1;
+		return arg[0] > 0 ? 1 : 0;
```

Comment:
I'm not entirely sure what the reasoning behind the old version here was...
But pinging @bshastry again about this: was this meant to basically simulate external calls pseudo-randomly failing or succeeding based on whether the passed in gas is odd or even?

We need to be careful about whether changing this makes fuzzing loose code paths after external calls.

Reply:
I don't remember doing this myself i.e., pseudo random return for call based on evenness of `arg[0]`

Since earlier we did not interpret call instructions, this might have been a way to simulate both success and failure events.

I don't quite understand the change. What is the semantics of the change proposed by the PR?